### PR TITLE
Feat simple moveit controller manager

### PR DESCRIPTION
**Pull request by ipa-hsd**
Currently the moveit config package generated by the Setup Assistant was not ready to use because of the missing controller configurations on the MoveIt! side. The PR generates a default `controllers.yaml` file along and sets the default `SimpleMoveItControllerManager`, which can be used within MoveIt! to communicate with `ros_control` or with the robot driver.

- f968db15ac6600af400c2044fd1dc8d5eafc2b5c - Currently `moveit_setup_assistant` generates an empty [`moveit_controller_manager.launch.xml`](https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_setup_assistant/templates/moveit_config_pkg_template/launch/moveit_controller_manager.launch.xml). I see this has been done in https://github.com/ros-planning/moveit/pull/1051, but it also does many other things like launching Gazebo.   

- c171dc9e531c5e8dd5f5c2fcd94ce7d03a22acc0 - Generates `controller.yaml` (MoveIt! specific, which would be loaded in the launch file above), with default `FollowJointTrajectory` action interface

- 19c13b7a48190260e188c2d4810271cbb1d8a722 - Introduced in https://github.com/ros-planning/moveit/pull/951, but this is specific to MoveIt!. The `controller_list` will be needed irrespective of `ros_control` or not.

### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Document API changes relevant to the user in the moveit/MIGRATION.md notes
- [ ] Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Decide if this should be cherry-picked to other current ROS branches
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_setup_assistant/include/moveit/setup_assistant/tools/moveit_config_data.h [hunk 1]
```diff
@@ -321,6 +321,7 @@ class MoveItConfigData
    * @param YAML Emitter - yaml emitter used to write the config to the ROS controllers yaml file
    * @param vector<ROSControlConfig> - a copy of ROS controllers config which will be modified in the function
    */
+  bool outputFollowJointTrajectoryYAML(const std::string& file_path);
```
Comment (davetcoleman) [inline moveit_setup_assistant/include/moveit/setup_assistant/tools/moveit_config_data.h]: add comment for how this is different than the second function below

**Comment by davetcoleman** [comment]
Can you better explain what this PR changes?

**Comment by ipa-hsd** [comment]
> Can you better explain what this PR changes?

Thanks for your feedback @davetcoleman. Could you please take a look again?

**Comment by MohmadAyman** [comment]
You did move the `controllers_list` from `ros_controllers` into a separate file `controllers.yaml` and sent it to the param server using `moveit_controller_manager.launch.xml`, correct?
I'm okay with these changes, but can we try to get  #1051 merged first, -it is also already approved- to not cause a conflict and then we can stack these changes on top?
@davetcoleman @ipa-hsd

**Comment by ipa-hsd** [comment]
@MohmadAyman wrote:
> You did move the controllers_list from ros_controllers into a separate file controllers.yaml and sent it to the param server using moveit_controller_manager.launch.xml, correct?  

Yes, that's correct.

> I'm okay with these changes, but can we try to get #1051 merged first, -it is also already approved- to not cause a conflict and then we can stack these changes on top?

That should be fine by me. Initially I thought `ros_control` is a hard dependency. But that does not seem to be the case.

**Comment by MohmadAyman** [comment]
@ipa-hsd #1051 has been merged, would you please rebase your branch so we can merge it :)

**Comment by rhaschke** [comment]
@ipa-hsd You was asked to rebased, but you merged kinetic-devel. This usually makes it more complicated to handle PRs on the maintainer side, because we cannot easily rebase anymore.

**Comment by ipa-hsd** [comment]
Sorry about that, was a mistake.

**Comment by henningkayser** [comment]
@ipa-hsd could you rebase this once more and target the master branch? I will do the second review in the meantime so we can get this merged.

**Comment by ipa-hsd** [comment]
https://github.com/ros-planning/moveit/pull/1402

**Comment by rhaschke** [comment]
Replaced by #1402.

**Pull request closed without merging**